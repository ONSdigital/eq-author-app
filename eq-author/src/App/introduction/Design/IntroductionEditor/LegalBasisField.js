import React from "react";
import PropTypes from "prop-types";

import { Input } from "components/Forms";
import {
  RadioLabel,
  RadioField,
  RadioTitle,
  RadioDescription,
} from "components/Radio";

export const LegalOption = ({ name, value, children, onChange, selected }) => (
  <RadioLabel htmlFor={value} selected={selected}>
    <Input
      type="radio"
      variant="radioBox"
      checked={selected}
      name={name}
      id={value}
      value={value}
      onChange={onChange}
    />
    {children}
  </RadioLabel>
);
LegalOption.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onChange: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired,
};

const OPTIONS = [
  {
    title: "Notice 1",
    notice: true,
    description:
      "Notice is given under section 1 of the Statistics of Trade Act 1947.",
    value: "NOTICE_1",
  },
  {
    title: "Notice 2",
    notice: true,
    description:
      "Notice is given under sections 3 and 4 of the Statistics of Trade Act 1947.",
    value: "NOTICE_2",
  },
  {
    title: "Voluntary",
    description: "No legal notice will be displayed.",
    value: "VOLUNTARY",
  },
];

const LegalBasisField = ({ name, value, onChange, ...rest }) => (
  <RadioField {...rest}>
    {OPTIONS.map((option) => (
      <LegalOption
        key={option.value}
        name={name}
        value={option.value}
        selected={option.value === value}
        onChange={onChange}
      >
        <RadioTitle>
          {`${option.title} ${
            option.notice ? "- Your response is legally required." : ""
          }`}
        </RadioTitle>
        <RadioDescription>{option.description}</RadioDescription>
      </LegalOption>
    ))}
  </RadioField>
);

LegalBasisField.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default LegalBasisField;
