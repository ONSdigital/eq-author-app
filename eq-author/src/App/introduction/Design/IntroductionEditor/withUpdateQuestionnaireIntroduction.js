import { graphql } from "react-apollo";
import gql from "graphql-tag";
import { filter } from "graphql-anywhere";

import ValidationErrorInfoFragment from "graphql/fragments/validationErrorInfo.graphql";

const mutation = gql`
  mutation UpdateQuestionnaireIntroduction(
    $input: UpdateQuestionnaireIntroductionInput!
  ) {
    updateQuestionnaireIntroduction(input: $input) {
      id
      title
      contactDetailsPhoneNumber
      contactDetailsEmailAddress
      contactDetailsEmailSubject
      contactDetailsIncludeRuRef
      additionalGuidancePanelSwitch
      additionalGuidancePanel
      description
      secondaryTitle
      secondaryDescription
      tertiaryTitle
      tertiaryDescription
      validationErrorInfo {
        ...ValidationErrorInfo
      }
    }
  }
  ${ValidationErrorInfoFragment}
`;
const inputFilter = gql`
  {
    id
    title
    contactDetailsPhoneNumber
    contactDetailsEmailAddress
    contactDetailsEmailSubject
    contactDetailsIncludeRuRef
    additionalGuidancePanelSwitch
    additionalGuidancePanel
    description
    secondaryTitle
    secondaryDescription
    tertiaryTitle
    tertiaryDescription
  }
`;

export const mapMutateToProps = ({ mutate }) => ({
  updateQuestionnaireIntroduction: (introduction) => {
    const data = filter(inputFilter, introduction);
    return mutate({
      variables: { input: data },
      optimisticResponse: {
        updateQuestionnaireIntroduction: {
          ...introduction,
          ...data,
          __typename: "QuestionnaireIntroduction",
        },
      },
    });
  },
});

export default graphql(mutation, {
  props: mapMutateToProps,
});
