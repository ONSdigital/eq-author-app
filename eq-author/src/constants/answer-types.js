export const CHECKBOX = "Checkbox";
export const RADIO = "Radio";
export const TEXTFIELD = "TextField";
export const TEXTAREA = "TextArea";
export const CURRENCY = "Currency";
export const NUMBER = "Number";
export const PERCENTAGE = "Percentage";
export const DATE = "Date";
export const DATE_RANGE = "DateRange";
export const UNIT = "Unit";
export const DURATION = "Duration";

export const ROUTING_ANSWER_TYPES = [
  RADIO,
  CURRENCY,
  NUMBER,
  PERCENTAGE,
  CHECKBOX,
  UNIT,
];

export const RADIO_OPTION = "RadioOption";
export const CHECKBOX_OPTION = "CheckboxOption";
export const MUTUALLY_EXCLUSIVE = "MutuallyExclusive";
export const MUTUALLY_EXCLUSIVE_OPTION = "MutuallyExclusiveOption";

export default [
  CHECKBOX,
  RADIO,
  TEXTFIELD,
  TEXTAREA,
  CURRENCY,
  NUMBER,
  PERCENTAGE,
  DATE,
  DATE_RANGE,
  UNIT,
  DURATION,
];
