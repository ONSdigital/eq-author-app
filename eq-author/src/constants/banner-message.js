const bannerMessageMigrated =
  "Author has been migrated and this page will turn off shortly.";
const bannerMessageVersion = "Please use latest version of Author.";
const bannerMessageQuestions =
  "If you have any questions, please contact the Author team.";

const bannerMessages = [
  bannerMessageMigrated,
  bannerMessageVersion,
  bannerMessageQuestions,
];

export default bannerMessages;
