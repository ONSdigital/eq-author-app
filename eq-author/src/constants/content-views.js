export const ANSWER = "answer";
export const METADATA = "metadata";

export default [ANSWER, METADATA];
