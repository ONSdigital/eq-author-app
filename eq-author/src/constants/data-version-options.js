const DATA_VERSION_OPTIONS = [
  {
    value: "1",
    title: "Data version 1",
    description:
      "Enables downstream processing of older questionnaires without complex features",
  },
  {
    value: "3",
    title: "Data version 3",
    description:
      "Enables downstream processing of newer questionnaires with complex features",
  },
];

export default DATA_VERSION_OPTIONS;
