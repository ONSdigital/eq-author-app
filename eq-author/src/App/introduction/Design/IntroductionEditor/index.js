import React from "react";
import gql from "graphql-tag";
import styled from "styled-components";
import { flowRight, noop } from "lodash/fp";
import { propType } from "graphql-anywhere";
import PropTypes from "prop-types";

import withPropRenamed from "enhancers/withPropRenamed";
import withChangeUpdate from "enhancers/withChangeUpdate";

import RichTextEditor from "components/RichTextEditor";
import withEntityEditor from "components/withEntityEditor";

import { colors } from "constants/theme";

import transformNestedFragments from "utils/transformNestedFragments";

import LegalBasisField from "./LegalBasisField";
import CollapsiblesEditor from "./CollapsiblesEditor";

import withUpdateQuestionnaireIntroduction from "./withUpdateQuestionnaireIntroduction";

const Section = styled.section`
  &:not(:last-of-type) {
    border-bottom: 1px solid #e0e0e0;
  }
`;

const Padding = styled.div`
  padding: 2em;
`;

const SectionTitle = styled.h2`
  font-size: 1.1em;
  font-weight: bold;
  color: ${colors.text};
  margin: 0 0 1em;
`;

const SectionDescription = styled.p`
  margin: 0.1em 0 1em;
  color: ${colors.textLight};
`;

const titleControls = {
  emphasis: true,
  piping: true,
};

const descriptionControls = {
  bold: true,
  emphasis: true,
  piping: true,
  list: true,
};

export const IntroductionEditor = ({ introduction, onChangeUpdate }) => {
  const {
    id,
    collapsibles,
    title,
    description,
    secondaryTitle,
    secondaryDescription,
    tertiaryTitle,
    tertiaryDescription,
    legalBasis,
  } = introduction;

  return (
    <>
      <Section>
        <Padding>
          <SectionTitle style={{ marginBottom: "0" }}>
            Introduction content
          </SectionTitle>
          <SectionDescription>
            This content is displayed above the “start survey” button. The title
            is not editable.
          </SectionDescription>
          <RichTextEditor
            id="intro-title"
            name="title"
            label="Title"
            multiline
            value={title}
            size="large"
            disabled
            onUpdate={noop}
            testSelector="txt-intro-title"
          />
          <RichTextEditor
            id="intro-description"
            name="description"
            label="Description"
            multiline
            value={description}
            controls={descriptionControls}
            onUpdate={onChangeUpdate}
            testSelector="txt-intro-description"
          />
          <SectionTitle>Legal basis</SectionTitle>
          <LegalBasisField
            name="legalBasis"
            value={legalBasis}
            onChange={onChangeUpdate}
            data-test="intro-legal-basis"
          />
        </Padding>
      </Section>
      <Section>
        <Padding>
          <SectionTitle style={{ marginBottom: "0" }}>
            Secondary content
          </SectionTitle>
          <SectionDescription>
            This content is displayed below the “start survey” button.
          </SectionDescription>
          <RichTextEditor
            id="secondary-title"
            name="secondaryTitle"
            label="Title"
            value={secondaryTitle}
            controls={titleControls}
            size="large"
            onUpdate={onChangeUpdate}
            testSelector="txt-intro-secondary-title"
          />
          <RichTextEditor
            id="secondary-description"
            name="secondaryDescription"
            label="Description"
            multiline
            value={secondaryDescription}
            controls={descriptionControls}
            onUpdate={onChangeUpdate}
            testSelector="txt-intro-secondary-description"
          />
          <SectionTitle style={{ marginBottom: "0" }}>
            Collapsibles
          </SectionTitle>
          <SectionDescription>
            Information which is displayed in a collapsible “twistie”.
          </SectionDescription>
          <CollapsiblesEditor introductionId={id} collapsibles={collapsibles} />
        </Padding>
      </Section>
      <Section>
        <Padding>
          <SectionTitle>Tertiary content</SectionTitle>
          <RichTextEditor
            id="tertiary-title"
            name="tertiaryTitle"
            label="Title"
            value={tertiaryTitle}
            controls={titleControls}
            size="large"
            onUpdate={onChangeUpdate}
            testSelector="txt-intro-tertiary-title"
          />
          <RichTextEditor
            id="tertiary-description"
            name="tertiaryDescription"
            label="Description"
            multiline
            value={tertiaryDescription}
            controls={descriptionControls}
            onUpdate={onChangeUpdate}
            testSelector="txt-intro-tertiary-description"
          />
        </Padding>
      </Section>
    </>
  );
};
const fragment = gql`
  fragment IntroductionEditor on QuestionnaireIntroduction {
    id
    title
    description
    secondaryTitle
    secondaryDescription
    collapsibles {
      ...CollapsibleEditor
    }
    tertiaryTitle
    tertiaryDescription
    legalBasis
  }
`;

IntroductionEditor.fragments = [fragment, ...CollapsiblesEditor.fragments];

IntroductionEditor.propTypes = {
  introduction: propType(
    transformNestedFragments(fragment, CollapsiblesEditor.fragments)
  ).isRequired,
  onChangeUpdate: PropTypes.func.isRequired,
};

const withWrappers = flowRight(
  withUpdateQuestionnaireIntroduction,
  withPropRenamed("updateQuestionnaireIntroduction", "onUpdate"),
  withEntityEditor("introduction"),
  withChangeUpdate
);

export default withWrappers(IntroductionEditor);
