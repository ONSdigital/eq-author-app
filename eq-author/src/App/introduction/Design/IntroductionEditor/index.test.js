import React from "react";
import { shallow } from "enzyme";

import IntroductionEditor from "./";

import { introductionErrors } from "constants/validationMessages";
import { useParams } from "react-router-dom";
import { useMutation } from "@apollo/react-hooks";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: jest.fn(),
  useMatch: jest.fn(),
  useParams: jest.fn(),
}));

jest.mock("@apollo/react-hooks", () => ({
  useMutation: jest.fn(() => [() => null]),
}));

describe("IntroductionEditor", () => {
  let props;
  beforeEach(() => {
    props = {
      introduction: {
        id: "1",
        title: "title",
        contactDetailsPhoneNumber: "0300 1234 931",
        contactDetailsEmailAddress: "surveys@ons.gov.uk",
        contactDetailsEmailSubject: "Change of details",
        contactDetailsIncludeRuRef: false,
        additionalGuidancePanelSwitch: false,
        previewQuestions: false,
        disallowPreviewQuestions: true,
        additionalGuidancePanel: "additionalGuidancePanel",
        description: "description",
        secondaryTitle: "secondary title",
        secondaryDescription: "secondary description",
        collapsibles: [],
        tertiaryTitle: "tertiary title",
        tertiaryDescription: "tertiary description",
        validationErrorInfo: {
          errors: [],
          totalCount: 0,
        },
        comments: [{}],
      },
      onChangeUpdate: jest.fn(),
      updateQuestionnaireIntroduction: jest.fn(),
    };

    useParams.mockImplementation(() => ({
      questionnaireId: "questionnaire-1",
      introductionId: "1",
    }));
  });

  const { PHONE_NOT_ENTERED, EMAIL_NOT_ENTERED } = introductionErrors;

  it("should render", () => {
    const propsWithPreviewQuestions = {
      ...props,
      introduction: {
        ...props.introduction,
        previewQuestions: true,
      },
    };
    expect(
      shallow(<IntroductionEditor {...propsWithPreviewQuestions} />)
    ).toMatchSnapshot();
  });

  it("should toggle the additional guidance panel", () => {
    const mockUseMutation = jest.fn();
    useMutation.mockImplementationOnce(jest.fn(() => [mockUseMutation]));

    const wrapper = shallow(<IntroductionEditor {...props} />);
    expect(
      wrapper.find('[name="additionalGuidancePanel"]').exists()
    ).toBeFalsy();
    wrapper
      .find("#toggle-additional-guidance-panel")
      .simulate("change", { target: { checked: true } });
    expect(mockUseMutation).toHaveBeenCalledTimes(1);
  });

  it("should show the additional guidance panel", () => {
    props.introduction.additionalGuidancePanelSwitch = true;
    const wrapper = shallow(<IntroductionEditor {...props} />);
    expect(
      wrapper.find('[name="additionalGuidancePanel"]').exists()
    ).toBeTruthy();
  });

  it("should toggle the add RU ref to the subject line", () => {
    const mockUseMutation = jest.fn();
    useMutation.mockImplementationOnce(jest.fn(() => [mockUseMutation]));

    const wrapper = shallow(<IntroductionEditor {...props} />);
    expect(
      wrapper.find('[name="toggle-contact-details-include-ruref"]').exists()
    ).toBeTruthy();
    wrapper
      .find("#toggle-contact-details-include-ruref")
      .simulate("change", { target: { checked: false } });
    expect(mockUseMutation).toHaveBeenCalledTimes(1);
  });

  it("should display validation error message when phone number is not entered", () => {
    props.introduction.validationErrorInfo = {
      errors: [
        {
          id: "1",
          errorCode: "ERR_VALID_REQUIRED",
          field: "contactDetailsPhoneNumber",
          type: "introduction",
        },
      ],
      totalCount: 1,
    };

    const wrapper = shallow(<IntroductionEditor {...props} />);

    expect(wrapper.find({ children: PHONE_NOT_ENTERED })).toHaveLength(1);
  });

  it("should display validation error message when email address is not entered", () => {
    props.introduction.validationErrorInfo = {
      errors: [
        {
          id: "1",
          errorCode: "ERR_VALID_REQUIRED",
          field: "contactDetailsEmailAddress",
          type: "introduction",
        },
      ],
      totalCount: 1,
    };

    const wrapper = shallow(<IntroductionEditor {...props} />);

    expect(wrapper.find({ children: EMAIL_NOT_ENTERED })).toHaveLength(1);
  });

  it("should update contact details", () => {
    const mockUseMutation = jest.fn();
    useMutation.mockImplementationOnce(jest.fn(() => [mockUseMutation]));

    const wrapper = shallow(<IntroductionEditor {...props} />);

    const phoneInput = wrapper.find(
      "[data-test='txt-contact-details-phone-number']"
    );
    const emailAddressInput = wrapper.find(
      "[data-test='txt-contact-details-email-address']"
    );
    const emailSubjectInput = wrapper.find(
      "[data-test='txt-contact-details-email-subject']"
    );

    phoneInput.simulate("blur", { target: { value: "0" } });
    emailAddressInput.simulate("blur", {
      target: { value: "test@test.com" },
    });
    emailSubjectInput.simulate("blur", {
      target: { value: "Test" },
    });

    expect(mockUseMutation).toHaveBeenCalledTimes(3);
  });

  //TODO: previewQuestions;
  describe("previewingQuestions", () => {
    it("Should render previewQuestions section", () => {
      const wrapper = shallow(<IntroductionEditor {...props} />);
      expect(wrapper.find('[name="previewQuestions-section"]')).toBeTruthy();
    });

    it("Should render previewQuestions toggle in off state", () => {
      const wrapper = shallow(<IntroductionEditor {...props} />);
      expect(
        wrapper.find('[name="toggle-preview-questions"]').prop("checked")
      ).toBe(false);
    });

    it("Should toggle preview questions", () => {
      const mockUseMutation = jest.fn();
      useMutation.mockImplementationOnce(jest.fn(() => [mockUseMutation]));

      const wrapper = shallow(<IntroductionEditor {...props} />);
      expect(
        wrapper.find('[name="toggle-preview-questions"]').exists()
      ).toBeTruthy();
      wrapper
        .find("#toggle-preview-questions")
        .simulate("change", { target: { checked: false } });
      expect(mockUseMutation).toHaveBeenCalledTimes(1);
    });

    it("Should render the warning panel when previewQuestions is true", () => {
      const propsWithPreviewQuestions = {
        ...props,
        introduction: {
          ...props.introduction,
          previewQuestions: true,
        },
      };
      const wrapper = shallow(
        <IntroductionEditor {...propsWithPreviewQuestions} />
      );
      expect(
        wrapper.find('[data-testid="preview-questions-warn-panel"]').exists()
      ).toBeTruthy();
    });

    it("Should render the disallow info panel when disallowPreviewQuestions is true", () => {
      const propsWithDisallowPreviewQuestions = {
        ...props,
        introduction: {
          ...props.introduction,
          disallowPreviewQuestions: true,
        },
      };
      const wrapper = shallow(
        <IntroductionEditor {...propsWithDisallowPreviewQuestions} />
      );
      expect(
        wrapper.find('[data-testid="preview-questions-info-panel"]').exists()
      ).toBeTruthy();
    });

    it("should not render the warning panel when previewQuestions is false", () => {
      const propsWithPreviewQuestions = {
        ...props,
        introduction: {
          ...props.introduction,
          previewQuestions: false,
        },
      };
      const wrapper = shallow(
        <IntroductionEditor {...propsWithPreviewQuestions} />
      );
      expect(
        wrapper.find('[data-testid="preview-questions-warn-panel"]').exists()
      ).toBeFalsy();
    });
  });
});
