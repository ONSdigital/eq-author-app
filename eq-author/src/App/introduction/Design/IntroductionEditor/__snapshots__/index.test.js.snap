// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroductionEditor should render 1`] = `
<Fragment>
  <IntroductionHeader />
  <IntroductionEditor__Section>
    <IntroductionEditor__Padding>
      <IntroductionEditor__SectionTitle
        style={
          Object {
            "marginBottom": "0",
            "marginTop": "-2em",
          }
        }
      >
        Introduction content
      </IntroductionEditor__SectionTitle>
      <IntroductionEditor__SectionDescription>
        This content is displayed above the “start survey” button. The title is not editable.
      </IntroductionEditor__SectionDescription>
      <RichTextEditor
        autoFocus={false}
        controls={
          Object {
            "bold": true,
            "emphasis": true,
            "heading": true,
            "link": true,
            "piping": true,
          }
        }
        disabled={false}
        errorValidationMsg={null}
        id="intro-title"
        label="Title"
        maxHeight={12}
        multiline={true}
        name="title"
        onUpdate={[Function]}
        placeholder=""
        size="large"
        testSelector="txt-intro-title"
        value="title"
        withoutMargin={true}
      />
      <Panel
        data-testid="introduction-content-info-panel"
        variant="warning"
        withList={false}
        withPanelMargin={true}
      >
        You can have this page display on the Hub via the 
        <Link
          to="/q/questionnaire-1/settings"
        >
          Settings page
        </Link>
      </Panel>
      <div>
        <IntroductionEditor__HorizontalSeparator />
        <IntroductionEditor__SectionTitle
          style={
            Object {
              "marginBottom": "0",
            }
          }
        >
          ONS contact details
        </IntroductionEditor__SectionTitle>
        <IntroductionEditor__SectionDescription>
          For business to report a change to company details or structure.
        </IntroductionEditor__SectionDescription>
        <Field
          disabled={false}
          last={false}
        >
          <Label
            bold={true}
            htmlFor="contactDetailsPhoneNumber"
          >
            Phone Number
          </Label>
          <IntroductionEditor__StyledInput
            data-test="txt-contact-details-phone-number"
            hasError={false}
            id="contactDetailsPhoneNumber"
            onBlur={[Function]}
            onChange={[Function]}
            value="0300 1234 931"
          />
        </Field>
        <Field
          disabled={false}
          last={false}
        >
          <Label
            bold={true}
            htmlFor="contactDetailsEmailAddress"
          >
            Email Address
          </Label>
          <IntroductionEditor__StyledInput
            data-test="txt-contact-details-email-address"
            hasError={false}
            id="contactDetailsEmailAddress"
            onBlur={[Function]}
            onChange={[Function]}
            value="surveys@ons.gov.uk"
          />
        </Field>
        <Field
          disabled={false}
          last={false}
        >
          <Label
            bold={true}
            htmlFor="contactDetailsEmailSubject"
          >
            Email Subject
          </Label>
          <withChangeHandler(UncontrolledInput)
            data-test="txt-contact-details-email-subject"
            id="contactDetailsEmailSubject"
            onBlur={[Function]}
            onChange={[Function]}
            value="Change of details"
          />
        </Field>
        <IntroductionEditor__InlineField
          open={false}
          style={
            Object {
              "marginBottom": "0",
            }
          }
        >
          <Label
            bold={true}
            htmlFor="toggle-contact-details-include-ruref"
          >
            Add RU ref to the subject line
          </Label>
          <ToggleSwitch
            blockDisplay={false}
            checked={false}
            hideLabels={false}
            id="toggle-contact-details-include-ruref"
            name="toggle-contact-details-include-ruref"
            onChange={[Function]}
          />
        </IntroductionEditor__InlineField>
        <IntroductionEditor__SectionDescription>
          Add the reporting unit reference to the end of the subject line, for example, Change of details reference 621476278652.
        </IntroductionEditor__SectionDescription>
        <IntroductionEditor__HorizontalSeparator />
      </div>
      <IntroductionEditor__InlineField
        open={false}
      >
        <Label
          bold={true}
          htmlFor="toggle-additional-guidance-panel"
        >
          Additional guidance panel
        </Label>
        <ToggleSwitch
          blockDisplay={false}
          checked={false}
          hideLabels={false}
          id="toggle-additional-guidance-panel"
          name="toggle-additional-guidance-panel"
          onChange={[Function]}
        />
      </IntroductionEditor__InlineField>
      <RichTextEditor
        autoFocus={false}
        controls={
          Object {
            "bold": true,
            "emphasis": true,
            "link": true,
            "list": true,
            "piping": true,
          }
        }
        disabled={false}
        id="intro-description"
        label="Description"
        maxHeight={12}
        multiline={true}
        name="description"
        onUpdate={[Function]}
        placeholder=""
        testSelector="txt-intro-description"
        value="description"
      />
      <IntroductionEditor__SectionTitle>
        Legal basis
      </IntroductionEditor__SectionTitle>
      <Panel
        variant="info"
        withLeftBorder={true}
        withList={false}
        withPanelMargin={true}
      >
        The legal basis can be changed on the Settings page
      </Panel>
    </IntroductionEditor__Padding>
  </IntroductionEditor__Section>
  <IntroductionEditor__Section
    name="previewQuestions-section"
  >
    <IntroductionEditor__Padding>
      <IntroductionEditor__InlineField
        disabled={true}
        style={
          Object {
            "marginBottom": "0",
          }
        }
      >
        <Label
          bold={true}
          labelFor="toggle-preview-questions"
          style={
            Object {
              "color": "#7a7a7a",
              "marginBottom": "0",
            }
          }
        >
          Include a link to a page for previewing the questions
        </Label>
        <ToggleSwitch
          blockDisplay={false}
          checked={true}
          hideLabels={false}
          id="toggle-preview-questions"
          name="toggle-preview-questions"
          onChange={[Function]}
        />
      </IntroductionEditor__InlineField>
      <IntroductionEditor__SectionDescription
        style={
          Object {
            "color": "#7a7a7a",
            "marginBottom": "1em",
          }
        }
      >
        Each section is represented as a collapsible element, allowing users to show and hide its respective questions.
      </IntroductionEditor__SectionDescription>
      <Panel
        data-testid="preview-questions-warn-panel"
        name="preview-questions-warn-panel"
        variant="warning"
        withList={false}
        withPanelMargin={true}
      >
        Adding a collection list will automatically turn off and disable this setting.
      </Panel>
    </IntroductionEditor__Padding>
  </IntroductionEditor__Section>
  <IntroductionEditor__Section>
    <IntroductionEditor__Padding>
      <IntroductionEditor__SectionTitle
        style={
          Object {
            "marginBottom": "0",
          }
        }
      >
        Secondary content
      </IntroductionEditor__SectionTitle>
      <IntroductionEditor__SectionDescription>
        This content is displayed below the “start survey” button.
      </IntroductionEditor__SectionDescription>
      <RichTextEditor
        autoFocus={false}
        controls={
          Object {
            "emphasis": true,
            "piping": true,
          }
        }
        disabled={false}
        id="secondary-title"
        label="Title"
        maxHeight={12}
        multiline={false}
        name="secondaryTitle"
        onUpdate={[Function]}
        placeholder=""
        size="large"
        testSelector="txt-intro-secondary-title"
        value="secondary title"
      />
      <RichTextEditor
        autoFocus={false}
        controls={
          Object {
            "bold": true,
            "emphasis": true,
            "link": true,
            "list": true,
            "piping": true,
          }
        }
        disabled={false}
        id="secondary-description"
        label="Description"
        maxHeight={12}
        multiline={true}
        name="secondaryDescription"
        onUpdate={[Function]}
        placeholder=""
        testSelector="txt-intro-secondary-description"
        value="secondary description"
      />
      <IntroductionEditor__SectionTitle
        style={
          Object {
            "marginBottom": "0",
          }
        }
      >
        Collapsibles
      </IntroductionEditor__SectionTitle>
      <IntroductionEditor__SectionDescription>
        Information which is displayed in a collapsible “twistie”.
      </IntroductionEditor__SectionDescription>
      <Apollo(Apollo(CollapsiblesEditor))
        collapsibles={Array []}
        introductionId="1"
      />
    </IntroductionEditor__Padding>
  </IntroductionEditor__Section>
  <IntroductionEditor__Section>
    <IntroductionEditor__Padding>
      <IntroductionEditor__SectionTitle>
        Tertiary content
      </IntroductionEditor__SectionTitle>
      <RichTextEditor
        autoFocus={false}
        controls={
          Object {
            "emphasis": true,
            "piping": true,
          }
        }
        disabled={false}
        id="tertiary-title"
        label="Title"
        maxHeight={12}
        multiline={false}
        name="tertiaryTitle"
        onUpdate={[Function]}
        placeholder=""
        size="large"
        testSelector="txt-intro-tertiary-title"
        value="tertiary title"
      />
      <RichTextEditor
        autoFocus={false}
        controls={
          Object {
            "bold": true,
            "emphasis": true,
            "link": true,
            "list": true,
            "piping": true,
          }
        }
        disabled={false}
        id="tertiary-description"
        label="Description"
        maxHeight={12}
        multiline={true}
        name="tertiaryDescription"
        onUpdate={[Function]}
        placeholder=""
        testSelector="txt-intro-tertiary-description"
        value="tertiary description"
      />
    </IntroductionEditor__Padding>
  </IntroductionEditor__Section>
</Fragment>
`;
